0001-typedb/02-dev/04-write.md

```diff
@@ -0,0 +1,685 @@
+---
+pageTitle: Writing data
+keywords: typeql, typedb, query, insert, delete, update, write
+longTailKeywords: typeql insert, typeql write data, typeql delete, typeql update
+Summary: Writing data in a TypeDB database.
+---
+
+# Writing data
+
+There are multiple query types that can modify data in a database with `write` transaction:
+
+- insert query
+- delete query
+- update (or match-delete-insert query)
+
+<div class="note">
+[Note]
+To modify the schema of a database, use [define](02-schema.md#define)/[undefine](02-schema.md#undefine) queries instead.
+</div>
+
+To try the following examples use [TypeDB Studio](../../02-clients/01-studio.md) or one of the other TypeDB 
+[Clients](../../02-clients/00-clients.md).
+
+For those developing applications with [TypeDB drivers](../../02-clients/00-clients.md#typedb-drivers), please see the 
+instructions and examples for a specific language/framework: 
+[Java](../../02-clients/03-java.md), 
+[Node.js](../../02-clients/05-nodejs.md), 
+[Python](../../02-clients/04-python.md).
+
+## Insert query
+
+An insert query, optionally preceded by a [match](03-match.md) clause, adds data (e.g., entities, relations and 
+attributes) to a database.
+
+### Syntax
+
+Insert queries are written in TypeQL with the following syntax:
+
+<!-- test-ignore -->
+```typeql
+[match <pattern>]
+insert <pattern>
+```
+
+The optional `match` clause uses a pattern to find existing data which is needed to insert new data. 
+For example, to insert a new relation, we need to match every entity, attribute, and/or other relation that will 
+play a role in it to be able to address them in the `insert` clause. 
+
+The `insert` clause uses patterns to specify the data to be inserted and may include existing data found by 
+the `match` clause.
+
+<div class="note">
+[Note]
+A pattern consists of variables and constraints for concepts. For more information see the 
+[Pattern syntax](03-match.md#pattern-syntax) section.
+</div>
+
+The `insert` query is the only query that can be without a `match` clause. In that case the pattern in the `insert` 
+clause will be used exactly once per query. 
```

Comment:
"the only data query" or smth like that? `define`/`undefine` don't need `match` either.

Reply:
```suggestion
The `insert` query is the only data query that can be without a `match` clause. In that case the pattern in the `insert` 
clause will be used exactly once per query. 
```